[pkg/ottl] Added ottl functions for hashing strings (#22968)
open-telemetry/opentelemetry-collector-contrib: pkg/ottl/ottlfuncs/README.md

- `ConvertCase(metric.name, "snake")`

### FNV

`FNV(value)`

The `FNV` Converter converts the `value` to an FNV hash/digest.

The returned type is int64.

> The input `value` types:
> * string. If it fails then nil will be returned.
> 
> If `value` is another type or parsing failed nil is always returned.
> 
> The `value` is either a path expression to a telemetry field to retrieve or a literal.

Reviewer: ```suggestion
`value` is either a path expression to a string telemetry field or a literal string. If `value` is another type an error is returned.

If an error occurs during hashing it will be returned.
```

Examples:

- `FNV(attributes["device.name"])`

> - `FNV("name")`

Author: @TylerHelmuth should these functions be upper or lower case? The same for `SHA1` and `SHA256`?

Reviewer: I think all uppercase is good

- `ParseJSON(body)`

### SHA1

`SHA1(value)`

The `SHA1` Converter converts the `value` to a sha1 hash/digest.

> The returned type is int64.
> 
> The input `value` types:
> * string. If it fails then nil will be returned.
> 
> If `value` is another type or parsing failed nil is always returned.
> 
> The `value` is either a path expression to a telemetry field to retrieve or a literal.

Reviewer: ```suggestion
The returned type is string.

`value` is either a path expression to a string telemetry field or a literal string. If `value` is another type an error is returned.

If an error occurs during hashing it will be returned.
```

Examples:

- `SHA1(attributes["device.name"])`

- `SHA1("name")`

### SHA256

`SHA256(value)`

The `SHA256` Converter converts the `value` to a sha256 hash/digest.

> The returned type is int64.
> 
> The input `value` types:
> * string. If it fails then nil will be returned.
> 
> If `value` is another type or parsing failed nil is always returned.
> 
> The `value` is either a path expression to a telemetry field to retrieve or a literal.

Reviewer: ```suggestion
The returned type is string.

`value` is either a path expression to a string telemetry field or a literal string. If `value` is another type an error is returned.

If an error occurs during hashing it will be returned.
```
